### PR TITLE
Arbitrary topology extension 

### PR DESCRIPTION
**Pull request by aryamangupta**
I have tested few topologies in this implementation. The 'bridges' folder contains tested tree network under switch named 's1' (with depth -  3 and fanout - 2).  CIDR allocation has to be adjusted, which I am still working on.

### CODE REVIEW

**Comment by cjdoucette** [comment]
Congratulations on completing Google Summer of Code! I would like to start the review process to have your code integrated into our main XLXC repository. Once your code is added, I think it will be of great benefit to that project.

I have a few questions:

You mentioned that "CIDR allocation has to be adjusted, which I am still working on." Have you finished working on this aspect? Is there anything else that needs to be added?

Also, do you have any other documentation for your project? In particular, it would be helpful to see how you tested the code. I see that you included a ReadmeSwitch.md file and TopologyReadMe.md file, along with an example of the bridges. However, those only describe the process, not show the process. It would be more helpful to have something like this to show the commands that are used to create the network:

https://github.com/AltraMayor/XIA-for-Linux/wiki/XIA-Linux-Containers-%28XLXC%29#Create_and_Use_a_Container_Network

In other words, if someone just downloaded your version of XLXC without knowing OVS, what would they do?

Please send along this kind of documentation, and in the meantime I will read through your code. Again congratulations, and thank you for your hard work!
